spec/integration/rspec/raise_error_matcher_spec.rb

```diff
@@ -741,4 +741,430 @@
+                      expected "#<a kind of StandardError and having attributes (message: #<a kind of String>)>"
+                    end
+                  end
+              )
+
+            expect(program).to produce_output_when_run(
+              expected_output
+            ).in_color(color_enabled)
+          end
+        end
+      end
+    end
+  end
+
+  context "with an RSpec matcher and expected message" do
+    context "when the block raises a different error than what is given" do
+      context "when the message is short" do
+        it "produces the correct failure message" do
+          as_both_colored_and_uncolored do |color_enabled|
+            snippet = <<~TEST.strip
+              expect { raise StandardError.new('boo') }.to raise_error(a_kind_of(RuntimeError), "boo")
+            TEST
+            program =
+              make_plain_test_program(snippet, color_enabled: color_enabled)
+
+            expected_output =
+              build_expected_output(
+                color_enabled: color_enabled,
+                snippet:
+                  "expect { raise StandardError.new('boo') }.to raise_error(a_kind_of(RuntimeError), \"boo\")",
+                expectation:
+                  proc do
+                    line do
+                      plain "Expected raised exception "
+                      actual %|#<StandardError "boo">|
+                      plain " to match "
+                      expected "#<a kind of RuntimeError \"boo\">"
+                      plain "."
+                    end
+                  end
+              )
+
+            expect(program).to produce_output_when_run(
+              expected_output
+            ).in_color(color_enabled)
+          end
+        end
+      end
+
+      context "when the message is long" do
+        it "produces the correct failure message" do
+          as_both_colored_and_uncolored do |color_enabled|
+            snippet = <<~TEST.strip
+              expect { raise StandardError.new('this is a super super super long message') }.to raise_error(a_kind_of(RuntimeError), "beep")
+            TEST
+            program =
+              make_plain_test_program(snippet, color_enabled: color_enabled)
+
+            expected_output =
+              build_expected_output(
+                color_enabled: color_enabled,
+                snippet:
+                  "expect { raise StandardError.new('this is a super super super long message') }.to raise_error(a_kind_of(RuntimeError), \"beep\")",
+                newline_before_expectation: true,
+                expectation:
+                  proc do
+                    line do
+                      plain "Expected raised exception "
+                      actual %|#<StandardError "this is a super super super long message">|
+                    end
+
+                    line do
+                      plain "                 to match "
+                      expected "#<a kind of RuntimeError \"beep\">"
```

Comment:
This admittedly reads a little weird, and I imagine would be even more so with compound matcher descriptions. Let me know if you have any ideas (or think it's OK).

Reply:
Hmm. This looks okay, although to be consistent with other ways that we do this, I'm wondering if it might make more sense to surround the arguments with parentheses:

```
#<a kind of RuntimeError ("beep")>
```

Reply:
@mcmire I just looked into this. The `"beep"` in this case is coming from the second positional argument to `raise_error`, not the first, if that makes sense. Something like `raise_error(a_kind_of(RuntimeError), "beep")`. Not sure if that makes a difference. 

Surrounding the expected message with parens is easy, but would also affect the existing formatting of expected errors that aren't matchers, like this:
```
raise_error(StandardError, "message")
# before: expected XYZ to match #<StandardError "message">
# after:  expected XYZ to match #<StandardError ("message")>
```

I could alternatively do some special casing to preserve existing behavior and add parens when the first arg is a matcher only. Let me know what you think!